### PR TITLE
refactor parallel installer

### PR DESCRIPTION
**Pull request by pducks32**
**Sorry I didn't have great Internet last week so thats why I didn't push this. I made some changes and it works on my system.** *Fingers Crossed*

So I was going through the Installer code and thought it was confusing and hard to read (and I guess CodeClimate agreed) and I thought that the parallel code was especially complex. So to get the class more closely aligned with SOLID and Sandi Metz's principles I ripped out that part into a new section.

Parallel code is very hard to test and I'm not always sure about whether things work on older versions of Ruby so i'm just as interested as anyone on how the tests turn out. I'm going to keep researching the best way to test parallelization but for right now I just left it.

### CODE REVIEW

**Comment by pducks32** [comment]
Is there anything I can do to get rid of the merge conflicts. Should I just refork and go again?

**Comment by indirect** [comment]
If you either rebase against master or merge the current master into this branch, the pull request should become mergable.

**Comment by pducks32** [comment]
I don't know whether to laugh or just -_- the fact that this finally passed and travis didn't just randomly fail.

**Comment by indirect** [comment]
Awesome! I'm very excited about this. :D Can you squash this all down to one commit, and force push to your master branch? Then I'll merge!

**Comment by pducks32** [comment]
Yea so wasn't totally sure how to do that and now I think I totally messed it up. Now all my commits are tangled with the ones from bundler/master and not all after each other. So how do I squash them properly?

**Comment by segiddins** [comment]
@pducks32 there are a bunch of answers: http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

File: lib/bundler/installer/parallel_installer.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    def dependencies
+      @dependencies ||= all_dependencies.reject {|dep| ignorable_dependency? dep }
+    end
+
+    def all_dependencies
+      @spec.dependencies
+    end
+  end
+
+  def self.call(*args)
```
Comment (segiddins) [inline lib/bundler/installer/parallel_installer.rb]: why not make these `call` methods `install`?

**Comment by segiddins** [comment]
also, @pducks32, it would be awesome if you could add some API docs for the new class!

**Pull request closed without merging**